integrations/webhook/src/config.tsx

```diff
@@ -0,0 +1,140 @@
+        switch (action.action) {
+            case 'save.config': {
+                const { api, environment } = context;
+                const spaceInstallation = environment.spaceInstallation!;
+                const webhookUrl = element.state.webhookUrl.trim();
+
+                // Validate webhook URL
+                if (!webhookUrl) {
+                    throw new ExposableError('Webhook URL is required');
+                }
+
+                // Very basic URL format validation
+                try {
+                    new URL(webhookUrl);
+                } catch (error) {
+                    throw new ExposableError('Please enter a valid webhook URL');
+                }
+
+                // Validate that at least one event is enabled
+                const hasEnabledEvents = EVENT_TYPES.some((eventType) => element.state[eventType]);
+                if (!hasEnabledEvents) {
+                    throw new ExposableError('At least one event type must be enabled');
+                }
+
+                await api.integrations.updateIntegrationSpaceInstallation(
+                    spaceInstallation.integration,
+                    spaceInstallation.installation,
+                    spaceInstallation.space,
+                    {
+                        configuration: {
+                            webhookUrl: webhookUrl,
+                            secret: element.state.secret,
+                            ...Object.fromEntries(
+                                EVENT_TYPES.map((eventType) => [
+                                    eventType,
+                                    element.state[eventType],
+                                ]),
+                            ),
+                        } as WebhookConfiguration,
+                    },
+                );
+
+                return { type: 'complete' };
+            }
+        }
+    },
+    render: async (element) => {
+        return (
+            <configuration>
+                <input
+                    label="Webhook URL"
+                    hint={<text>The URL where the events will be sent.</text>}
+                    element={
+                        <textinput
+                            state="webhookUrl"
+                            placeholder="https://your-domain.com/webhook"
+                        />
+                    }
+                />
+
+                <divider size="medium" />
+
+                <text>Select which events you want to receive:</text>
+
+                {EVENT_TYPES.map((eventType) => (
+                    <input
+                        label={AVAILABLE_EVENTS[eventType]}
+                        element={<switch state={eventType} />}
+                    />
+                ))}
+
+                <divider size="medium" />
+
+                <markdown content="### Secret" />
+                <text>
+                    You can copy and verify this secret in your webhook for HMAC verification:
```

Comment:
```suggestion
                    Use this secret to verify the signature included in GitBook webhook events (i.e HMAC with SHA-256 verification):
```

We can maybe at some stage link to a section in our docs that talks about webhooks (how to verify signature and example of events objects)

Reply:
Yes, good idea 💯 
Most of this stuff is fairly standard, but giving some examples it's definitely a lot nicer. I'll add some docs.